inc/Engine/Optimization/RUCSS/Warmup/ResourceFetcher.php

```diff
@@ -0,0 +1,196 @@
+<?php
+declare(strict_types=1);
+
+namespace WP_Rocket\Engine\Optimization\RUCSS\Warmup;
+
+use WP_Rocket\Engine\Optimization\AssetsLocalCache;
+use WP_Rocket\Engine\Optimization\RegexTrait;
+use WP_Rocket\Engine\Optimization\UrlTrait;
+use WP_Rocket\Logger\Logger;
+use WP_Rocket_WP_Async_Request;
+
+class ResourceFetcher extends WP_Rocket_WP_Async_Request {
+
+	use RegexTrait, UrlTrait;
+
+	/**
+	 * Prefix
+	 *
+	 * @var string
+	 */
+	protected $prefix = 'rocket';
+
+	/**
+	 * Action
+	 *
+	 * @var string
+	 */
+	protected $action = 'saas_warmup';
+
+	/**
+	 * Resources array.
+	 *
+	 * @var array
+	 */
+	private $resources = [];
+
+	/**
+	 * Assets local cache instance
+	 *
+	 * @var AssetsLocalCache
+	 */
+	private $local_cache;
+
+	/**
+	 * Resource fetcher process instance.
+	 *
+	 * @var ResourceFetcherProcess
+	 */
+	private $process;
+
+	/**
+	 * Resource constructor.
+	 *
+	 * @param AssetsLocalCache       $local_cache Local cache instance.
+	 * @param ResourceFetcherProcess $process     Resource fetcher process instance.
+	 */
+	public function __construct( AssetsLocalCache $local_cache, ResourceFetcherProcess $process ) {
+		$this->local_cache = $local_cache;
+		$this->process     = $process;
+		parent::__construct();
```

Comment:
Probably move parent::__construct(); before the param initialization and add a space after it.
```
parent::__construct();

$this->local_cache = $local_cache;
$this->process     = $process;
```